brave/brave-ios
# Fix #7623: Native asset is not added as visible asset once a custom network is added

## Summary of Changes
When user adds a custom network via dapp request or wallet settings, we will need to store it's native asset as the default visible asset. For the same reason, once user decides to delete a custom network, we will delete all the local stored assets that are in this network.


This pull request fixes #7623

## Submitter Checklist:


## Test Plan:

1. Add a custom network from chainlist.org (connect wallet then add network)
2. approve the request, switch/not switch to that network 
3. go to portfolio
4. observe the new network's native asset has been automatically added as a visible asset
5. Manually add a custom network from settings 
6. go to portfolio 
7. observe the new network's native asset has been automatically added as a visible asset 
8. Remove a custom network from settings
9. go to portfolio 
10. observe all the assets that in the removed network have been removed. 
11. Navigate to the [CAROT add token request page](https://vittominacori.github.io/watch-token/page/?hash=0x7b2261646472657373223a22307831666635316461373163336435623866383531313265613361343434306265623637613632616439222c226c6f676f223a22227d&network=mainnet). 
12. Tap 'Add to MetaMask' button, and accept the request to add the suggested token in wallet
13. go to portfolio
14. observe the suggested token has been added as a visible assets

## Screenshots:


https://github.com/brave/brave-ios/assets/1187676/7abd5624-411b-4474-bfad-cac9118b3a66



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
